plugins/coral-plugin-respect/client/components/RespectButton.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import styles from './style.css';
+import Icon from './Icon';
+
+import {I18n} from 'coral-framework';
+import cn from 'classnames';
+import translations from '../translations.json';
+
+const lang = new I18n(translations);
+
+export default class RespectButton extends React.Component {
+
+  handleClick = () => {
+    const {postRespect, showSignInDialog, deleteAction, commentId} = this.props;
+    const {me, comment} = this.props.data;
+
+    const respect = comment.action_summaries[0];
+    const respected = (respect && respect.current_user);
+
+    // If the current user does not exist, trigger sign in dialog.
+    if (!me) {
+      const offset = document.getElementById(`c_${commentId}`).getBoundingClientRect().top - 75;
```

Comment:
this probably needs to be wrapped in a `try / catch` since we're calling a method after querying the DOM.